pdl_syntax.md

```diff
@@ -0,0 +1,828 @@
+   longAnswer: string
+ ]
+}
+```
+In the above example, the union answerFormat has three members, with two string type members differentiated using the aliases (shortAnswer and longAnswer). When aliases are used, the alias becomes the "member key" for the union members and will be used in the wire format. 
+
+## Enum Type
+
+Enums types may contain any number of symbols. 
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+enum Fruits {
+  APPLE
+  BANANA
+  ORANGE
+  PINEAPPLE
+}
+```
+
+Enums can be referenced in other schemas by name.
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: Fruits
+}
+```
+
+Enums can also be referenced in other schemas by inlining their type definition. 
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: enum Fruits { APPLE, BANANA, ORANGE }
+} 
+```
+
+### Enum documentation, deprecation and properties
+Doc comments, @deprecation and properties can be added directly to enum symbols. 
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A fruit
+ */
+enum Fruits {
+
+  @color = "red"
+  APPLE
+
+  /**
+   * A yummy fruit.
+   */
+  @color = "yellow"
+  BANANA
+
+  @deprecated
+  @color = "orange"
+  ORANGE
+}
+```
+
+### Enum defaults
+To specify defaults, specify the enum value as string.
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: enum Fruits { APPLE, BANANA, ORANGE } = "APPLE"
+} 
+```
+
+
+The default value can also be defined as following exmaple:
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: Fruits = "APPLE"
+} 
+```
+
+## Typerefs
+Pegasus supports a new schema type known as a typeref. A typeref is like
+a typedef in C. It does not declare a new type but declares an alias to
+an existing type.
+
+  - Typerefs are useful for differentiating different uses of the same
+    type. For example, we can use to a typeref to differentiate a string
+    field that holds an URN (uniform resource name) from an arbitrary
+    string value or a long field that holds an epoch time in
+    milliseconds from a generic long value.
+  - A typeref allows additional meta-data to be associated with
+    primitive and unnamed types. This meta-data can be used to provide
+    documentation or support custom properties.
+  - A typeref provides a way to refer to common unnamed types such as
+    arrays, maps, and unions. Without typerefs, users may have to wrap
+    these unnamed types with a record in order to address them.
+    Alternatively, users may cut-and-paste common type declarations,
+    resulting in unnecessary duplication and potentially causing
+    inconsistencies if future changes are not propagated correctly to
+    all copies.
+
+
+### Provide a name for a union, map, or array. So that it can be referenced by name。
```

Comment:
These headers seem odd because there's no context. It's as if they're supposed to be in a list of "reasons to use typerefs". You could either make a section that's explicit about this, or you can just reword the headers to something like `Typerefs can be used to name anonymous types`, `Typerefs can provide additional clarity when using primitive types`, and `Typerefs can be used to specify custom types and coercers` (though this last one should probably be moved to another Java-specific page, perhaps you can do that later).

Reply:
updated the headers